### PR TITLE
Output deploy config and use in preview mode to locate config files

### LINKED ISSUE #88: Integrate with wrangler deploy redirect

### PR DESCRIPTION
**Pull request by jamesopstad**
resolves #88

### CODE REVIEW

File: packages/vite-plugin-cloudflare/src/deploy-config.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	auxiliaryWorkers: Array<{ configPath: string }>;
+}
+
+function getDeployConfigPath(root: string) {
+	return path.resolve(root, '.wrangler', 'deploy', 'config.json');
+}
+
+export function getConfigPaths(root: string) {
+	const deployConfigPath = getDeployConfigPath(root);
+	const deployConfig = JSON.parse(
```
Comment (petebacondarwin) [inline packages/vite-plugin-cloudflare/src/deploy-config.ts:18]: Might be safer to use JSONC or JSON5 parser that can handle JSON comments.

File: packages/vite-plugin-cloudflare/src/deploy-config.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+		fs.readFileSync(deployConfigPath, 'utf-8'),
+	) as DeployConfig;
+
+	return [
+		{ configPath: deployConfig.configPath },
```
Comment (petebacondarwin) [inline packages/vite-plugin-cloudflare/src/deploy-config.ts:23]: Just for fun you could actually do:

```suggestion
		deployConfig,
```

File: packages/vite-plugin-cloudflare/src/deploy-config.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+				);
+
+				const configPath = path.relative(
+					deployConfigDirectory,
+					path.resolve(
+						resolvedViteConfig.root,
+						outputDirectory,
+						'wrangler.json',
+					),
+				);
```
Comment (petebacondarwin) [inline packages/vite-plugin-cloudflare/src/deploy-config.ts:79]: This is duplicated above. Perhaps put in a little helper?

**Review by petebacondarwin — APPROVED** [review]

File: packages/vite-plugin-cloudflare/src/deploy-config.ts (hunk 3 above)
Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/deploy-config.ts]: [8bdd014](https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/122/commits/8bdd014c1a3edb8de3a5bab1392d8c3f367519b0)

File: packages/vite-plugin-cloudflare/src/deploy-config.ts (hunk 1 above)
Comment (jamesopstad) [reply packages/vite-plugin-cloudflare/src/deploy-config.ts:18]: I'll leave it for now, as we control the output, but we can change this later if necessary.

**Comment by jamesopstad** [comment]
I made an additional change to not resolve the plugin config (and read the input Worker configs) in preview mode. Will merge now as will be off for a couple of weeks.

**Pull request merged**